proto/http.c

```diff
@@ -301,14 +301,35 @@ static int http_parse(struct wsgi_request *wsgi_req, char *watermark) {
 		}
 	}
 	else {
-		// TODO add ipv6 support
-		memset(ip, 0, INET_ADDRSTRLEN+1);
-		if (inet_ntop(AF_INET, (void *) &http_sin->sin_addr.s_addr, ip, INET_ADDRSTRLEN)) {
-			wsgi_req->len += proto_base_add_uwsgi_var(wsgi_req, "REMOTE_ADDR", 11, ip, strlen(ip));
-		}
-		else {
-			uwsgi_error("inet_ntop()");
-			return -1;
+		// TODO log something useful for AF_UNIX sockets
+		switch (http_sa->sa_family) {
+			case AF_INET6:
+				{
+					memset(ip, 0, INET6_ADDRSTRLEN+1);
```

Comment:
you should use `sizeof(ip)` in the last field, it's less fragile if the size of `ip` changes in the future